### ISSUE TITLE
Library scan terminates on audio files with empty tags

### ISSUE THREAD

**Issue by sscraggles** [DobyTang/LazyLibrarian#928]
### LazyLibrarian version number (at the bottom of config page)
latest today, 9b5daf82e4f1a54fa192f671e79acb42e4330f7e
### Operating system used (windows, mac, linux, NAS type)
Debian 9
### Interface in use (default, bookstrap)
Bookstrap
### Which api (Goodreads, GoogleBooks, both)
Goodreads
### Source of your LazyLibrarian installation (git, zip file, 3rd party package)
Git clone
### Relevant debug log with api keys and any passwords redacted

`20-Jul-2017 22:58:09 - DEBUG   :: AUDIO_SCAN : [/media/Audio Books] Now scanning subdirectory /Terence McKenna/True Hallucinations`
`
20-Jul-2017 22:58:09 - ERROR   :: AUDIO_SCAN : Unhandled exception in libraryScan: Traceback (most recent call last):
  File "/opt/LazyLibrarian/lazylibrarian/librarysync.py", line 556, in LibraryScan
    newauthor, authorid, new = addAuthorNameToDB(author)  # get the author name as we know it...
  File "/opt/LazyLibrarian/lazylibrarian/importer.py", line 38, in addAuthorNameToDB
    if len(author) < 2:
TypeError: object of type 'NoneType' has no len()
`

**Comment by philborman** [comment]
Ah, the old tag reader returned an empty string, seems this new one returns None.
Easy to fix.

**Comment by philborman** [comment]
Ok, fixed the bug now, can cope with no tags. The next problem is that if there are no tags, the only way to get the title/author of the audiobook is by trying to pattern match the filename. There might be a text file of some sort with the mp3 files giving info?  Maybe we should ignore audiobooks without tags and ask the user to add them?

**Comment by sscraggles** [comment]
Dunno, does the system mandate one book per directory, or support a flat file structure with multiple books per directory.  I was curious to discover that tags are even read to be honest, I'd thought it might just look for presence of audio files in an exptected directory pattern.  Many books can have multiple files with spurious tags.
But then, I don't plan to use it for downloading and importing, so my usecase is probably limited.

**Comment by sscraggles** [comment]
Tested fine with ddb9d30e4ac9327d02fce276a91f6e6d72e6b1f1 .  Log reports pattern match failed and library scan continues.

**Comment by seanvree** [comment]
I'm having a similar issue, didn't want to open a new issue just yet.

OS:  Windows 10E X64 1703
Python 2.7.14
LL v: c072296f24279f172c7342abd774100f3581e057

Getting this error:

```
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:  03-Oct-2017 04:34:03 - ERROR   :: AUDIO_SCAN : librarysync.py:LibraryScan:917 : Unhandled exception in libraryScan: Traceback (most recent call last):
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:    File "c:\lazylibrarian\lazylibrarian\librarysync.py", line 764, in LibraryScan
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:      book_filename = audioRename(bookid)
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:    File "c:\lazylibrarian\lazylibrarian\bookwork.py", line 41, in audioRename
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:      r = os.path.dirname(book_filename)
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:    File "C:\Python27\lib\ntpath.py", line 215, in dirname
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:      return split(p)[0]
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:    File "C:\Python27\lib\ntpath.py", line 180, in split
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:      d, p = splitdrive(p)
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:    File "C:\Python27\lib\ntpath.py", line 115, in splitdrive
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:      if len(p) > 1:
Oct 03 04:34:03 LazyLibrarian lazylibrarian.log:  TypeError: object of type 'NoneType' has no len()
```

**Comment by philborman** [comment]
There is some incomplete testing code in the latest version that I'm waiting on feedback for. 
If you turn off "Rename existing books" in config the problem will go away. Should be a proper fix in a day or two. Sorry about that.

**Comment by seanvree** [comment]
@philborman  right, I only got that error when I turned that function on. 

Thanks!

**Comment by seanvree** [comment]
@philborman  I think this is fixed now.

**Comment by seanvree** [comment]
@philborman  was this ever fixed?  I thought it was but still getting the error

I"m on master 2c751da410801306afbd517f6f55fb84bfdd637f :


```

2017-10-10 09:48:58 | ERROR | Unhandled exception in libraryScan: Traceback (most recent call last): File "c:\lazylibrarian\lazylibrarian\librarysync.py", line 715, in LibraryScan book_filename = bookRename(bookid) File "c:\lazylibrarian\lazylibrarian\bookwork.py", line 170, in bookRename r = os.path.dirname(f) File "C:\Python27\lib\ntpath.py", line 215, in dirname return split(p)[0] File "C:\Python27\lib\ntpath.py", line 180, in split d, p = splitdrive(p) File "C:\Python27\lib\ntpath.py", line 115, in splitdrive if len(p) > 1: TypeError: object of type 'NoneType' has no len()
-- | -- | --



```

**Comment by philborman** [comment]
It was fixed, but this is not the same bug. I managed to break it again when I added the bookRename function. Will push a fix shortly.
